SteamKit2/SteamKit2/Steam/WebAPI/WebAPI.cs

```diff
@@ -210,6 +210,14 @@ public sealed class AsyncInterface : DynamicObject, IDisposable
                 RegexOptions.Compiled | RegexOptions.IgnoreCase
             );
 
+            /// <summary>
+            /// Gets or sets the URI which is used to execute WebAPI requests.
+            /// When overriden, "secure" option is ignored. This also affects internal SteamKit requests.
+            /// </summary>
+            /// <value>
+            /// Endpoint URI which is used to execute WebAPI requests. For example, "http://steam-api.local/proxy".
+            /// </value>
+            public static string HttpEndpoint { get; set; }
```

Comment:
Hm, what are your thoughts on adding properties to `SmartCMServerList` and having it pass it through to the `SteamDirectory` calls?

Are there any other places within SK that you need to override the endpoint? I believe `SteamDirectory` is likely the only outlier, but I might be wrong.